pyproject.toml

```diff
@@ -177,24 +177,24 @@ ruff = "^0.0.250"
 optional = true
 
 [tool.poetry.group.test.dependencies]
-coverage = { version = "^7.3", extras = ["toml"] }
+coverage = { version = ">=7.3,<8", extras = ["toml"] }
 dash = { version = "^2.6", extras = ["testing"] }
-dirty-equals = "^0.5.0"
-freezegun = "^1.2"
-ipykernel = "^6.19.4"
-jsonschema = { version = "^4.17.3", extras = ["format-nongpl"] }
-jupyter = "^1.0.0"
+dirty-equals = ">=0.5.0,<1"
+freezegun = ">=1.2,<2"
+ipykernel = ">=6.19.4,<7"
+jsonschema = { version = ">=4.17.3,<5", extras = ["format-nongpl"] }
+jupyter = ">=1,<2"
 lmfit = "^1.1.0"  # required for example observations_station_gaussian_model.py
-percy-selenium = "^2.0.0"
-pybufrkit = "^0.2"
-pytest = "^7.2"
-pytest-cov = "^4.0"
-pytest-dictsdiff = "^0.5"
-pytest-notebook = "^0.8"
-pytest-xdist = "^3"
-selenium = "^4.0"
+percy-selenium = ">=2,<3"
+pybufrkit = ">=0.2,<0.3"
+pytest = ">=7.2,<8"
+pytest-cov = ">=4,<5"
+pytest-dictsdiff = ">=0.5,<0.6"
+pytest-notebook = ">=0.8,<0.9"
+pytest-xdist = ">=3,<4"
+selenium = ">=4,<5"
 surrogate = "^0.1"
-webdriver-manager = "^4.0.0"
+webdriver-manager = ">=4,<5"
```

Comment:
Let me use this as an example to demonstrate why I prefer the version range notation: When version `5` of this package will be released, Dependabot will submit a change to `>=4,<6`, and the outcome from CI will immediately tell whether this is good or not. If version 5 works, and just because it is version 5 already, I am trusting their maintainers enough that it will not break during the whole version 5 release cycle.

On the other hand, a "point" bump would raise the lower version bound of this package to ">=5", which is an unneccessary restriction if it still works with version 4.

Those thoughts include a series of assumptions and decisions I would like to make individually per dependency. They hopefully outline my thoughts why I am so much in favor of version range notations. Of course, it only explains my thoughts on behalf of a single example, but I am sure you can extrapolate from that.